app/assets/javascripts/oxalis/view/right-menu/comment_tab/tree_comment_list.js

```diff
@@ -25,30 +26,51 @@ type State = {
 };
 
 class TreeCommentList extends React.PureComponent<TreeCommentListProps, State> {
+  treeDomElement: ?HTMLLIElement;
+  activeNodeHasComment: boolean;
   state = {
     collapsed: false,
   };
 
+  componentDidUpdate() {
+    this.ensureVisible();
+  }
+
+  ensureVisible() {
+    // Only scroll to this tree if this is the active tree and there is no active comment to scroll to
+    if (
+      this.props.tree.treeId === this.props.skeletonTracing.activeTreeId &&
+      !this.activeNodeHasComment
+    ) {
+      scrollIntoViewIfNeeded(this.treeDomElement, { centerIfNeeded: true });
+    }
+  }
+
   handleToggleComment = () => {
     this.setState({ collapsed: !this.state.collapsed });
   };
 
   render() {
     const containsActiveNode = this.props.tree.treeId === this.props.skeletonTracing.activeTreeId;
+    this.activeNodeHasComment = false;
```

Comment:
This is a side effect in the render function. I am surprised that the linter didn't complain. I would strongly urge against including side effects in the render function. There are other life cycle hooks for precisely that, e.g. `componentWillUpdate`.

Reply:
You're right, not my proudest PR. I addressed the issue in my last commit.